extensions/Worker.Extensions.Tables/src/TypeConverters/TableObjectConverter.cs

```diff
@@ -0,0 +1,103 @@
+    /// Converter to bind <see cref="object" /> type parameters.
+    /// </summary>
+    [SupportsDeferredBinding]
+    internal class TableObjectConverter : TableConverterBase<object>
+    {
+        private readonly IOptions<WorkerOptions> _workerOptions;
+
+        public TableObjectConverter(IOptions<WorkerOptions> workerOptions, IOptionsMonitor<TablesBindingOptions> tableOptions, ILogger<TableObjectConverter> logger)
+            : base(tableOptions, logger)
+        {
+            _workerOptions = workerOptions ?? throw new ArgumentNullException(nameof(workerOptions));
+        }
+
+        public override async ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            try
+            {
+                if (!CanConvert(context))
+                {
+                    return ConversionResult.Unhandled();
+                }
+
+                var modelBindingData = context?.Source as ModelBindingData;
+                var tableData = GetBindingDataContent(modelBindingData);
+                var result = await ConvertModelBindingData(tableData, context?.TargetType);
+
+                if (result is null)
+                {
+                    return ConversionResult.Failed(new InvalidOperationException($"Unable to convert table binding data to type '{context.TargetType.Name}'."));
+                }
+
+                return ConversionResult.Success(result);
+            }
+            catch (Exception ex)
+            {
+                return ConversionResult.Failed(ex);
+            }
+        }
+
+        private async Task<object?> ConvertModelBindingData(TableData content, Type targetType)
+        {
+            if (string.IsNullOrEmpty(content.TableName))
+            {
+                throw new ArgumentNullException(nameof(content.TableName));
+            }
+
+            if (string.IsNullOrEmpty(content.PartitionKey))
+            {
+                throw new ArgumentNullException(nameof(content.PartitionKey));
+            }
+
+            var tableClient = GetTableClient(content.Connection, content.TableName!);
+
+            if (targetType.IsCollectionType())
+            {
+                IEnumerable<TableEntity> tableEntities = await GetEnumerableTableEntity(content);
+                return DeserializeToTargetObjectAsync(targetType, tableEntities);
+            }
+            else 
+            {
+                if (string.IsNullOrEmpty(content.RowKey))
+                {
+                    throw new ArgumentNullException(nameof(content.RowKey));
+                }
+
+                TableEntity tableEntity = await tableClient.GetEntityAsync<TableEntity>(content.PartitionKey, content.RowKey);
+                return DeserializeToTargetObjectAsync(targetType, tableEntity);
+            }
+        }
+
+        private object? DeserializeToTargetObjectAsync(Type targetType, object tableEntity)
+        {
+            string jsonString = JsonSerializer.Serialize(tableEntity);
```

Comment:
I noticed we're using the STJ serializer here, but on line 100 we use whatever serializer is configured. Could this lead to any issues? Try testing by configuring the app to use Json.NET and check if this method still behaves as expected.

Reply:
Tried with NewtonSoft serializer too. It's working fine. Added tests to cover these scenarios.